### PR TITLE
Add Hash instance for NonEmptyMap.

### PR DESCRIPTION
**Pull request by takayahilton**

### CODE REVIEW

File: core/src/main/scala/cats/data/NonEmptyMapImpl.scala [hunk 1]
```diff
@@ -318,10 +318,8 @@ sealed abstract private[data] class NonEmptyMapInstances {
         NonEmptyList(fa.head._2, fa.tail.toList.map(_._2))
     }
 
-  implicit def catsDataEqForNonEmptyMap[K: Order, A: Eq]: Eq[NonEmptyMap[K, A]] =
-    new Eq[NonEmptyMap[K, A]] {
-      def eqv(x: NonEmptyMap[K, A], y: NonEmptyMap[K, A]): Boolean = x === y
-    }
+  implicit def catsDataHashForNonEmptyMap[K: Hash: Order, A: Hash]: Hash[NonEmptyMap[K, A]] =
+    Hash[SortedMap[K, A]].asInstanceOf[Hash[NonEmptyMap[K, A]]]
```
Comment (LukaJCB) [inline core/src/main/scala/cats/data/NonEmptyMapImpl.scala]: We should consider adding a `Contravariant[Hash]`, so we don't need to `asInstanceOf` these, WDYT @kailuowang ?

Comment (kailuowang) [reply core/src/main/scala/cats/data/NonEmptyMapImpl.scala]: Contravariant[Hash] sounds reasonable, but that would add overhead to this particular case right. So maybe we add that in a separate PR?

Comment (LukaJCB) [reply core/src/main/scala/cats/data/NonEmptyMapImpl.scala]: Yes, just in general :)

Comment (kailuowang) [reply core/src/main/scala/cats/data/NonEmptyMapImpl.scala]: so shall we merge this one?

Comment (LukaJCB) [reply core/src/main/scala/cats/data/NonEmptyMapImpl.scala]: indeed

**Pull request merged**